keel-test/src/main/kotlin/com/netflix/spinnaker/keel/test/CombinedRepository.kt

```diff
@@ -1,30 +1,62 @@
+import com.netflix.spinnaker.keel.persistence.memory.InMemoryPausedRepository
 import com.netflix.spinnaker.keel.persistence.memory.InMemoryResourceRepository
 import io.mockk.mockk
 import java.time.Clock
+import org.springframework.context.ApplicationEventPublisher
+
+class InMemoryCombinedRepository(
+  override val clock: Clock = Clock.systemUTC(),
+  override val deliveryConfigRepository: InMemoryDeliveryConfigRepository = InMemoryDeliveryConfigRepository(clock),
+  override val artifactRepository: InMemoryArtifactRepository = InMemoryArtifactRepository(clock),
+  override val resourceRepository: InMemoryResourceRepository = InMemoryResourceRepository(clock),
+  val pausedRepository: InMemoryPausedRepository = InMemoryPausedRepository(clock),
+  private val eventPublisher: ApplicationEventPublisher = mockk(relaxed = true),
+  override val actuationPauser: ActuationPauser = ActuationPauser(resourceRepository, pausedRepository, eventPublisher, clock)
+) : CombinedRepository(
+  deliveryConfigRepository,
+  artifactRepository,
+  resourceRepository,
+  clock,
+  eventPublisher,
+  actuationPauser
+) {
+  fun dropAll() {
+    deliveryConfigRepository.dropAll()
+    artifactRepository.dropAll()
+    resourceRepository.dropAll()
+    pausedRepository.flush()
+  }
+}
 
 /**
  * A util for generating a combined repository with in memory implementations for tests
  */
 fun combinedInMemoryRepository(
   clock: Clock = Clock.systemUTC(),
-  deliveryConfigRepository: DeliveryConfigRepository = InMemoryDeliveryConfigRepository(clock),
-  artifactRepository: ArtifactRepository = InMemoryArtifactRepository(clock),
-  resourceRepository: ResourceRepository = InMemoryResourceRepository(clock)
-): CombinedRepository =
-  CombinedRepository(deliveryConfigRepository, artifactRepository, resourceRepository, clock, mockk(relaxed = true))
+  deliveryConfigRepository: InMemoryDeliveryConfigRepository = InMemoryDeliveryConfigRepository(clock),
+  artifactRepository: InMemoryArtifactRepository = InMemoryArtifactRepository(clock),
+  resourceRepository: InMemoryResourceRepository = InMemoryResourceRepository(clock),
+  pausedRepository: InMemoryPausedRepository = InMemoryPausedRepository(clock)
+): InMemoryCombinedRepository =
+  InMemoryCombinedRepository(clock, deliveryConfigRepository, artifactRepository, resourceRepository, pausedRepository)
 
 fun combinedMockRepository(
   deliveryConfigRepository: DeliveryConfigRepository = mockk(relaxed = true),
   artifactRepository: ArtifactRepository = mockk(relaxed = true),
   resourceRepository: ResourceRepository = mockk(relaxed = true),
-  clock: Clock = Clock.systemUTC()
+  pausedRepository: PausedRepository = mockk(relaxed = true),
+  clock: Clock = Clock.systemUTC(),
+  publisher: ApplicationEventPublisher = mockk(relaxed = true),
+  actuationPauser: ActuationPauser = ActuationPauser(resourceRepository, pausedRepository, publisher, clock)
```

Comment:
Another example of where putting service methods in `CombinedRepository` leads to weird patterns. Here, I'm adding a dependency on both `PausedRepository` and to `ActuationPauser` since the latter depends on the former, all so that I can add service methods that can query pause information. I had previously introduced a `ResourceService` that would be a new home for such service methods, but went back on it because I think we need to agree on a path forward as a team before I make further changes of that nature.